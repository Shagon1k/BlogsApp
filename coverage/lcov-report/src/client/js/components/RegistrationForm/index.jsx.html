<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/client/js/components/RegistrationForm/index.jsx</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="..\..\..\..\..\prettify.css" />
    <link rel="stylesheet" href="..\..\..\..\..\base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(..\..\..\..\..\sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="..\..\..\..\..\index.html">All files</a> / <a href="index.html">src/client/js/components/RegistrationForm</a> index.jsx
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">72.34% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>34/47</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">57.89% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>22/38</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">66.67% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>12/18</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">78.57% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>11/14</span>
      </div>
    </div>
  </div>
  <div class='status-line medium'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53
54
55
56
57
58
59
60
61
62
63</td><td class="line-coverage quiet"><span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">5x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React from 'react';<span class="cstat-no" title="statement not covered" ><span class="missing-if-branch" title="if path not taken" >I</span><span class="missing-if-branch" title="else path not taken" >E</span><span class="missing-if-branch" title="else path not taken" >E</span></span>
import { Link } from 'react-router-dom';
import './styles.scss';<span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="fstat-no" title="function not covered" ><span class="branch-1 cbranch-no" title="branch not covered" ><span class="missing-if-branch" title="else path not taken" >E</span><span class="branch-1 cbranch-no" title="branch not covered" ><span class="missing-if-branch" title="if path not taken" >I</span><span class="branch-2 cbranch-no" title="branch not covered" ><span class="branch-1 cbranch-no" title="branch not covered" ><span class="branch-0 cbranch-no" title="branch not covered" ><span class="missing-if-branch" title="if path not taken" >I</span><span class="missing-if-branch" title="if path not taken" >I</span></span></span></span></span></span></span></span></span></span></span></span></span></span>
&nbsp;
class RegistrationForm extends React.Component {
	constructor(props) {<span class="branch-1 cbranch-no" title="branch not covered" ></span>
		super(props);
&nbsp;
		this.state = {
			username: '',
			email: '',
			password: '',
			password2: ''
		}
	}<span class="fstat-no" title="function not covered" ></span>
&nbsp;
	handleInputChange(e) {
		const name =<span class="cstat-no" title="statement not covered" > e.target.name</span>;
		const value =<span class="cstat-no" title="statement not covered" > e.target.value</span>;
&nbsp;
		t<span class="cstat-no" title="statement not covered" >his.setState({</span>
			[name]: value
		})
	}
&nbsp;
	handleSubmit(e) {
		e.preventDefault();
		if (!this.state.username || !this.state.password || !this.state.password2) return;
		this.props.onRegister({
			username: this.state.username,
			email: this.state.email,
			password: this.state.password,
			password2: this.state.password2
		})
	}
&nbsp;
	render() {
		return (
			&lt;div className="registrationFormContainer"&gt;
				{this.props.isLoggedIn ? 
					&lt;span className="infoText"&gt;
						You are logged in!
					&lt;/span&gt; : 
					&lt;form className="registrationForm" onSubmit={(e) =&gt; this.handleSubmit(e)}&gt;
						&lt;label htmlFor="inputUsername"&gt;Username:&lt;/label&gt;
						&lt;input name="username" id="inputUsername" type="text" value={this.state.username} onChange={(e) =&gt; this.handleInputChange(e)} /&gt;<span class="fstat-no" title="function not covered" ></span>
						&lt;label htmlFor="inputEmail"&gt;Email:&lt;/label&gt;
						&lt;input name="email" id="inputEmail" type="text" value={this.state.email} onChange={(e) =&gt; this.handleInputChange(e)} /&gt;<span class="fstat-no" title="function not covered" ></span>
						&lt;label htmlFor="inputPassword"&gt;Password:&lt;/label&gt;
						&lt;input name="password" id="inputPassword" type="password" value={this.state.password} onChange={(e) =&gt; this.handleInputChange(e)} /&gt;<span class="fstat-no" title="function not covered" ></span>
						&lt;label htmlFor="inputPassword2"&gt;Confirm password:&lt;/label&gt;
						&lt;input name="password2" id="inputPassword2" type="password" value={this.state.password2} onChange={(e) =&gt; this.handleInputChange(e)} /&gt;<span class="fstat-no" title="function not covered" ></span>
						&lt;button type="submit" className="registerBtn btn"&gt;Register&lt;/button&gt;
					&lt;/form&gt;
				}
				&lt;Link className="lnk" to="/"&gt;Home Page&lt;/Link&gt;
			&lt;/div&gt;
		);
	}
}
&nbsp;
export default RegistrationForm;<span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="missing-if-branch" title="else path not taken" >E</span></span></span>
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Sat Mar 03 2018 15:03:49 GMT+0300 (Belarus Standard Time)
</div>
</div>
<script src="..\..\..\..\..\prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="..\..\..\..\..\sorter.js"></script>
</body>
</html>
