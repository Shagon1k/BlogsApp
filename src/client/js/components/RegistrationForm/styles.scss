.registrationFormContainer {
	position: absolute;
	max-width: 35%;
	height: 380px;
	box-sizing: border-box;
	padding: 20px;
	margin: auto;
	top: 0;
	bottom: 0;
	right: 0;
	left: 0;
	background: #50626C;
	color: white;
	display: flex;
	flex-flow: column wrap;
	align-items: center;
	justify-content: space-between;

	.infoText {
		font-size: 18px;
	}

	.registrationForm {
		display: flex;
		flex-flow: column wrap;
		align-items: center;
		justify-content: space-between;

		label {
			font-size: 18px;
			margin-bottom: 3px;
		}

		input,
		textarea {
			height: 25px;
			font-size: 18px;
			margin-bottom: 10px;
		}
		
		.registerBtn {
			width: 150px;
			height: 40px;
			border-radius: 5px;
			margin: 0 auto;
		}
	}
}