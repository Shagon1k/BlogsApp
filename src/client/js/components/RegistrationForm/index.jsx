import React from 'react';
import { Link } from 'react-router-dom';
import './styles.scss';

class RegistrationForm extends React.Component {
	constructor(props) {
		super(props);

		this.state = {
			username: '',
			email: '',
			password: '',
			password2: ''
		}
	}

	handleInputChange(e) {
		const name = e.target.name;
		const value = e.target.value;

		this.setState({
			[name]: value
		})
	}

	handleSubmit(e) {
		e.preventDefault();
		if (!this.state.username || !this.state.password || !this.state.password2) return;
		this.props.onRegister({
			username: this.state.username,
			email: this.state.email,
			password: this.state.password,
			password2: this.state.password2
		})
	}

	render() {
		return (
			<div className="registrationFormContainer">
				{this.props.isLoggedIn ? 
					<span className="infoText">
						You are logged in!
					</span> : 
					<form className="registrationForm" onSubmit={(e) => this.handleSubmit(e)}>
						<label htmlFor="inputUsername">Username:</label>
						<input name="username" id="inputUsername" type="text" value={this.state.username} onChange={(e) => this.handleInputChange(e)} />
						<label htmlFor="inputEmail">Email:</label>
						<input name="email" id="inputEmail" type="text" value={this.state.email} onChange={(e) => this.handleInputChange(e)} />
						<label htmlFor="inputPassword">Password:</label>
						<input name="password" id="inputPassword" type="password" value={this.state.password} onChange={(e) => this.handleInputChange(e)} />
						<label htmlFor="inputPassword2">Confirm password:</label>
						<input name="password2" id="inputPassword2" type="password" value={this.state.password2} onChange={(e) => this.handleInputChange(e)} />
						<button type="submit" className="registerBtn btn">Register</button>
					</form>
				}
				<Link className="lnk" to="/">Home Page</Link>
			</div>
		);
	}
}

export default RegistrationForm;
