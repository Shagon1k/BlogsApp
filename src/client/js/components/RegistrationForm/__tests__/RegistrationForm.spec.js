import React from 'react';
import RegistrationForm from '../index.jsx';

describe('RegistrationForm component', () => {
	const onRegister = jest.fn();

	it('renders "You are logged in!" if user is logged in', () => {
		const wrapper = shallow(<RegistrationForm isLoggedIn={true} onRegister={onRegister}/>);
		expect(wrapper).toMatchSnapshot();

	});
	
	it('renders register form if user is not logged in', () => {
		const wrapper = shallow(<RegistrationForm isLoggedIn={false} onRegister={onRegister}/>);
		expect(wrapper).toMatchSnapshot();
	})

	it('not handles submit if fields not filled event correctly', () => {
		const wrapper = shallow(<RegistrationForm isLoggedIn={false} onRegister={onRegister}/>);
		
		wrapper.find('.registrationForm').simulate('submit', { preventDefault() {} });
		expect(onRegister).not.toBeCalled();		
	})

	it('handles submit event correctly', () => {
		const wrapper = shallow(<RegistrationForm isLoggedIn={false} onRegister={onRegister}/>);
		
		wrapper.instance().setState({
			username : 'Ololo', 
			email: 'ololo@gmail.com', 
			password: 'password', 
			password2: 'password'
		}); 

		wrapper.find('.registrationForm').simulate('submit', { preventDefault() {} });
		expect(onRegister).toBeCalled();		
	})
})