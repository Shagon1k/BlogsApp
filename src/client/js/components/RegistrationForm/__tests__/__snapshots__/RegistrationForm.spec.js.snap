// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RegistrationForm component renders "You are logged in!" if user is logged in 1`] = `
<div
  className="registrationFormContainer"
>
  <span
    className="infoText"
  >
    You are logged in!
  </span>
  <Link
    className="lnk"
    replace={false}
    to="/"
  >
    Home Page
  </Link>
</div>
`;

exports[`RegistrationForm component renders register form if user is not logged in 1`] = `
<div
  className="registrationFormContainer"
>
  <form
    className="registrationForm"
    onSubmit={[Function]}
  >
    <label
      htmlFor="inputUsername"
    >
      Username:
    </label>
    <input
      id="inputUsername"
      name="username"
      onChange={[Function]}
      type="text"
      value=""
    />
    <label
      htmlFor="inputEmail"
    >
      Email:
    </label>
    <input
      id="inputEmail"
      name="email"
      onChange={[Function]}
      type="text"
      value=""
    />
    <label
      htmlFor="inputPassword"
    >
      Password:
    </label>
    <input
      id="inputPassword"
      name="password"
      onChange={[Function]}
      type="password"
      value=""
    />
    <label
      htmlFor="inputPassword2"
    >
      Confirm password:
    </label>
    <input
      id="inputPassword2"
      name="password2"
      onChange={[Function]}
      type="password"
      value=""
    />
    <button
      className="registerBtn btn"
      type="submit"
    >
      Register
    </button>
  </form>
  <Link
    className="lnk"
    replace={false}
    to="/"
  >
    Home Page
  </Link>
</div>
`;
